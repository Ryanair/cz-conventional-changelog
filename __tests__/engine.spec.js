describe('Engine', () => {
    let engine = require('../index');
    let answersMock = {
        type: 'chore',
        scope: 'buildspec',
        subject: 'This is a test message',
        body: '',
        isBreaking: false,
        needDeployment: true,
        isIssueAffected: true,
        issues: 'fix #Issue01'
    };

    let cz = {
        prompt: jest.fn()
    };

    beforeEach(() => {
        cz.prompt.mockReset();
    });

    test('should not add [skip-ci] to commit message head if user said no (default) to skip deployment', (done) => { 
        cz.prompt.mockReturnValue(Promise.resolve(answersMock));
        engine.prompter(cz, (commitMessage) => {
            expect(commitMessage).toMatchSnapshot();
            expect(/\[skip-ci\]/g.test(commitMessage)).toBeFalsy();
            done();
        });
    });

    test('should add [skip-ci] to commit message head if user said yes to skip deployment', (done) => { 
        cz.prompt.mockReturnValue(Promise.resolve({
            ...answersMock,
            needDeployment: false
        }));
        engine.prompter(cz, (commitMessage) => {
            expect(commitMessage).toMatchSnapshot();
            expect(/\[skip-ci\]/g.test(commitMessage)).toBeTruthy();
            done();
        });
    });
});