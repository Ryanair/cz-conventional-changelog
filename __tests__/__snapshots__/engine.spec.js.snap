// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engine should add [skip-ci] to commit message head if user said yes to skip deployment 1`] = `
"chore(buildspec): [skip-ci] This is a test message



fix #Issue01"
`;

exports[`Engine should not add [skip-ci] to commit message head if user said no (default) to skip deployment 1`] = `
"chore(buildspec): This is a test message



fix #Issue01"
`;
